### PR TITLE
Add CsVecBase::to_dense()

### PR DESCRIPTION
**Pull request by matzhaugen**
Add method `CsVecBase::to_dense()`

### CODE REVIEW

File: src/sparse/to_dense.rs [hunk 1]
```diff
@@ -75,4 +75,39 @@ mod test {
...
+    N: Clone,
+    I: SpIndex,
+{
+    if spvec.dim() != array.len() {
+        panic!("Dimension mismatch");
+    }
+
+    for (ind, val) in spvec.iter() {
+        array[[ind]] = val.clone();
+    }
```
Comment (vbarrielle) [inline src/sparse/to_dense.rs:112]: Bt convention, implementations are generally defined above the `test` module, which makes it easier to find them. Could you move that up?

**Review by vbarrielle — CHANGES_REQUESTED** [review]
Thanks for this implementation. It's looking good, I'd just like to have the implementation above the `test` module, and near `assign_to_dense`, to be able to easily browse the file. Thanks!

**Comment by vbarrielle** [comment]
Fixes #257.

File: src/sparse/to_dense.rs (hunk 1 above)
Comment (matzhaugen) [reply src/sparse/to_dense.rs]: Sure!

**Review by vbarrielle — APPROVED** [review]
All good now, thanks!

**Pull request merged**